libraries/state_history/include/eosio/state_history/log.hpp

```diff
@@ -263,6 +263,9 @@ class state_history_log {
          const uint32_t num_blocks_in_log = _end_block - _begin_block;
          fc::raw::pack(log, num_blocks_in_log);
       }
+
+      log.flush();
+      index.flush();
```

Comment:
Will this have any performance impact? But correctness is more important and this is only on ship node. It should be OK.

Do we have tests which repeatedly kill and shutdown ship?

Reply:
Yes, there is a performance impact. Likely fairly small.
We do not test `kill -9` with SHiP as there is no guarantee it will work. If the `kill -9` happens during or between the flushes then this will not help. However, during manual testing I was able to `kill -9` and recover from snapshot. Since most of the time is outside the write/flush this does help.